chain-signatures/node/src/protocol/presignature.rs

```diff
@@ -187,6 +187,17 @@ impl PresignatureManager {
 
     pub async fn insert(&mut self, presignature: Presignature) {
         tracing::debug!(id = ?presignature.id, "inserting presignature");
+        if self.contains(&presignature.id).await {
+            tracing::error!(id = presignature.id, "mine presignature already inserted");
+            return;
+        }
+        if self.contains_used(&presignature.id).await {
+            tracing::error!(
+                id = presignature.id,
+                "tried to insert used mine presignature"
+            );
+            return;
+        }
         // Remove from taken list if it was there
         self.gc.remove(&presignature.id);
```

Comment:
gc field is actually not necessary in PresignatureManager now that you add the used presignatures to storage. 
You can delete it. Same for TripleManager.

Reply:
GC is used for multiple purposes. I want to discuss that with @ChaoticTempest

Reply:
I will address GC in a separate PR.